### PR TITLE
chore(build): improve bump version script

### PR DESCRIPTION
**Pull request by jsdidierlaurent**
- add make version command (bump version and commit VERSION file)
- add make release command (add tag based on VERSION file)
- github-release script is based on VERSION file
- make run / make run-faker / make build are based on VERSION file

fix: #33

### CODE REVIEW

File: scripts/version/release [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+if [[ -n $(git status -s) ]]; then
+  echo "You have changed files. Plz clean up your git directory."
+  exit 1
+fi
+
+git checkout master
+git pull --rebase origin master
+
+if [[ -n $(git status | grep "nothing to commit, working directory clean") ]]; then
```
[the remark this answers is not in the document]
Comment (Alex-D) [reply scripts/version/release]: Also, we should maybe check that before run a rebase to avoid 💥

File: scripts/version/release [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  echo "You have unpushed commit(s). Plz sync your master branch to origin"
```
Comment (Alex-D) [inline scripts/version/release]: Plz => Please
Or
"Local is not sync with origin remote, please sync before release"

File: scripts/version/bump [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+function error {
+  echo -e "$1" >&2
+  exit 1
+}
+
+if [[ -n $(git status | grep 'Changes to be committed') ]]; then
+  echo "Your staging area is not empty. "git reset" will be executed"
+  read -p 'Continue ? (y|n): ' input
+  if [ $input == "y" ]; then
+    git reset && echo ""
```
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply scripts/version/bump]: It's for add \n after git reset comment. But it's not necessary anymore.

File: Makefile [hunk 4]
```diff
@@ -77,6 +77,15 @@ run: ## run monitoror
 run-faker: ## run monitoror in faker mode
 	@./scripts/run/faker
 
+# ============= VERSION =============
+.PHONY: version
+version: ## bump version of monitoror
+	@./scripts/version/bump
+
+.PHONY: release
+release: ## bump version of monitoror
```
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply Makefile]: Fixed

File: scripts/version/bump [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+VERSION_FILE=./VERSION
+
+function error {
+  echo -e "$1" >&2
+  exit 1
+}
+
+if [[ -n $(git status | grep 'Changes to be committed') ]]; then
+  echo "Your staging area is not empty. "git reset" will be executed"
+  read -p 'Continue ? (y|n): ' input
```
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply scripts/version/bump]: It's not necessary anymore.

File: scripts/version/bump [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  if [ $input == "y" ]; then
+    git reset && echo ""
+  else
+    exit 0
+  fi
+fi
+
+version=$(cat "$VERSION_FILE" 2>/dev/null)
+echo "Current Version $version"
```
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply scripts/version/bump]: Fixed

File: scripts/version/bump [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+set -o errexit -o nounset -o pipefail
+
+
+#######################################################################################
+# This script is used by "make version" command for creating new version of monitoror #
+#######################################################################################
+
+
+VERSION_REGEX="^(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)$"
```
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply scripts/version/bump]: This regex can detect wrong version like 01.1.1. But I changed anyway for fix version if this format is used.

File: scripts/version/bump [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    minor) new="${major}.$((minor + 1)).0";;
+    patch) new="${major}.${minor}.$((patch + 1))";;
+    *) error "invalid command $command" ;;
+  esac
+else
+  error "version $version does not match the version scheme 'major.minor.patch'."
+fi
+
+# Update VERSION file
+echo "Version $version -> $new"
```
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply scripts/version/bump]: Fixed

File: scripts/version/release [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+########################################################################################
+# This script is used by "make release" command for releasing new version of monitoror #
+########################################################################################
+
+
+VERSION_FILE=./VERSION
+
+if [[ -n $(git status -s) ]]; then
+  echo "You have changed files. Plz clean up your git directory."
```
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply scripts/version/release]: Fixed

File: scripts/version/release (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsdidierlaurent) [reply scripts/version/release]: Your link explain how to check if git is dirty. I already use this method. But i need to check if your master branch have commit to push.

So i change for :
```
if [[ $(git rev-parse origin/master) != $(git rev-parse HEAD) ]]; then
```

It's more "robust" to i18n

File: scripts/version/release (hunk 2 above)
Comment (jsdidierlaurent) [reply scripts/version/release]: Fixed

**Pull request merged**